frame/staking/src/lib.rs

```diff
@@ -1507,6 +1573,7 @@ decl_module! {
 			let stash = &ledger.stash;
 			ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
 			ensure!(targets.len() <= T::MAX_NOMINATIONS as usize, Error::<T>::TooManyTargets);
+			ensure!(NominatorsCount::get() < T::MaxNominators::get(), Error::<T>::TooManyValidators);
```

Comment:
You are not dealing with congestion here. 

Last brief talk with w3f was that it is okay to do something like: 

- you provide an `Option<T::AccountId>`, ONLY if the limit is reached, and the provided `T::AccountId` has a lower stash than you, then you can replace them.

Reply:
Also, this is wrong I think: If you are already a nominator, you should always be able to change your nomination. Only if you don't exist, this limit must hold.

```suggestion
			ensure!(Nominators::<T>::exists(stash).or_else(|| { NominatorsCount::get() < T::MaxNominators::get() }), Error::<T>::TooManyValidators);
```